jupyterhub/files/hub/jupyterhub_config.py

```diff
@@ -372,6 +372,11 @@ def camelCaseify(s):
     c.Spawner.debug = True
 
 
+# load seeded secrets
+c.JupyterHub.proxy_auth_token = get_secret_value("JupyterHub.proxy_auth_token")
+c.JupyterHub.cookie_secret = a2b_hex(get_secret_value("JupyterHub.cookie_secret"))
+c.CryptKeeper.keys = get_secret_value("CryptKeeper.keys").split(";")
+
 # load hub.config values
 for section, sub_cfg in get_config("hub.config", {}).items():
     c[section].update(sub_cfg)
```

Comment:
@minrk when we write `c.JupyterHub.cookie_secret = "asdf"` in lines above, will that make loading of a `hub.config.JupyterHub.cookie_secret` entry here fail?

Is the practice of assigning directly to a traitlet problematic if you want to work with it later like this?

Reply:
assigning directly to a traitlet is fine, but should only be done if there really is a value to assign (assigning None is **not** the same as not assigning). However, as you say, if hub.config.JupyterHub.cookie_secret is still defined, we are assigning to the same value, possibly with two different values (i.e. first, `a2b_hex(hex_cookie_secret)` and second the original (wrong) hex `hub.config.JupyterHub.cookie_secret`, in which case the above assignment will have no effect, as it will be replaced with the original value here the second time the same traitlet is assigned. If you want an to have higher priority, it must be done *last*, as it is assignment just like any other code:

```python
a = a2b_hex(value)
a = value
print(a) # will be `value`
```

Reply:
Ah, this would actually also fail unless we ensure to transform hub.config.JupyterHub.cookie_secret is triggered on modifying the traitlet value I figure.

I pushed 25bf8a0 to resolve this and created https://github.com/jupyterhub/jupyterhub/pull/3343 to avoid future struggles with cookie_secret in general.